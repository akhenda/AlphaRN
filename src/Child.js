import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';

import { capitalizeWords } from './utils/capitalize';

export default class Child extends Component {
  getCapitalizedText() {
    const { text } = this.props;
    if (text) return capitalizeWords(text);

    return 'You have not written anything!';
  }

  render() {
    return (
      <View testID='Child' style={styles.container}>
        <Text testID='ChildTitle' style={styles.title}>
          Capitalized Text:
        </Text>
        <Text testID='ChildOutput' style={styles.text}>
          {this.getCapitalizedText()}
        </Text>
        <TouchableOpacity testID='ClearText' onPress={this.props.onClear}>
          <Text style={styles.button}>
            CLEAR TEXT
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

Child.propTypes = {
  text: PropTypes.string.isRequired,
  onClear: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  container: {
    marginTop: 20,
    alignItems: 'center',
  },
  title: {
    fontSize: 18,
    marginBottom: 20,
  },
  text: {
    fontSize: 16,
    padding: 20,
  },
  button: {
    fontSize: 18,
    marginTop: 30,
    color: 'azure',
    borderRadius: 6,
    fontWeight: 'bold',
    paddingVertical: 10,
    paddingHorizontal: 15,
    backgroundColor: 'darkkhaki',
  },
});
