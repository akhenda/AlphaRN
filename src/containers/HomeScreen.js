import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Platform, Text, View, TextInput } from 'react-native';
import { connect } from 'react-redux';

import Child from 'src/components/Child';
import LoadingIndicator from 'src/components/LoadingIndicator';
import { getUserInfo } from 'src/state/actions/auth';
import styles from './styles/HomeScreenStyles';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' +
    'Cmd+D or shake for dev menu',
  android: 'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

class HomeScreen extends Component {  
  constructor(props) {
    super(props);

    this.state = { text: '' };
  }
  
  setText(text) {
    this.setState({ text });
  }
  
  handleTextChange(text) {
    this.setText(text);
  }

  clearText() {
    this.setText('');
  }

  render() {
    const { user } = this.props;
    const { text } = this.state;

    if (user === null) return <LoadingIndicator />;

    return (
      <View testID='Welcome' style={styles.container}>
        <Text testID='WelcomeTitle' style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text testID='WelcomeInstruction' style={styles.instructions}>
          To get started, edit App.js
        </Text>
        <Text testID='Instructions' style={styles.instructions}>
          {instructions}
        </Text>
        <TextInput
          value={text}
          testID='TextInput'
          style={styles.input}
          placeholder={'Write something...'}
          onChangeText={txt => this.handleTextChange(txt)}
        />
        <Child text={text} onClear={() => this.clearText()} />
      </View>
    );
  }
}

HomeScreen.propTypes = {
  user: PropTypes.object,
  token: PropTypes.string,
  getUserInfo: PropTypes.func,
};

const mapStateToProps = (state) => {
  return {
    user: state.auth.user,
    token: state.auth.token,
  };
};

export default connect(mapStateToProps, { getUserInfo })(HomeScreen);
