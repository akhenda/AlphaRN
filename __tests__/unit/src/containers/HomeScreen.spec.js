import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import HomeScreen from 'src/containers/HomeScreen';
import Child from 'src/components/Child';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('HomeScreen', () => {
  let component;
  let textInput;
  const defaultState = { text: '', loading: false };
  
  // Initialize mockstore
  const initialState = {
    auth: {
      user: null,
      token: '', 
    },
    app: {},
  };
  const store = mockStore(initialState);

  test('renders correctly', () => {
    // const tree = renderer.create(<HomeScreen store={store} />).toJSON();
    // expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    // component = shallow(<HomeScreen store={store} />);
    // textInput = component.find('TextInput');
  });

  it('has default state', () => {
    // expect(component.state()).toEqual(defaultState);
  });

  it('renders welcome text', () => {
    // const expectedText = 'Welcome to React Native!';
    // const text = component.find('Text').first().children().text();
    // expect(text).toEqual(expectedText);
  });

  it('renders input field with placeholder', () => {
    // const expectedPlaceholder = 'Write something...';
    // expect(textInput.length).toBe(1);
    // expect(textInput.props().placeholder).toEqual(expectedPlaceholder);
  });

  describe('when text changes', () => {
    // const newText = 'random string';

    beforeEach(() => {
      // textInput.simulate('changeText', newText);
    });

    it('updates component state', () => {
      // expect(component.state().text).toEqual(newText);
    });

    it('passes text from state to Child', () => {
      // const childComponent = component.find(Child);
      // expect(childComponent.props().text).toEqual(newText);
    });

    describe('when clearText callback is called', () => {
      beforeEach(() => {
        // const childComponent = component.find(Child);
        // childComponent.simulate('clear');
      });

      it('resets state', () => {
        // expect(component.state()).toEqual(defaultState);
      });
    });
  });
});
