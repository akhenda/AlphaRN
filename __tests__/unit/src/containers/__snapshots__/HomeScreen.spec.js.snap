// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="Welcome"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
    testID="WelcomeTitle"
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
    testID="WelcomeInstruction"
  >
    To get started, edit App.js
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
    testID="Instructions"
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Write something..."
    style={
      Object {
        "alignSelf": "center",
        "height": 60,
        "marginTop": 30,
        "padding": 10,
        "width": "80%",
      }
    }
    testID="TextInput"
    value=""
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 20,
      }
    }
    testID="Child"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
          "marginBottom": 20,
        }
      }
      testID="ChildTitle"
    >
      Capitalized Text:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "padding": 20,
        }
      }
      testID="ChildOutput"
    >
      You have not written anything!
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="ClearText"
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "darkkhaki",
            "borderRadius": 6,
            "color": "azure",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginTop": 30,
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        CLEAR TEXT
      </Text>
    </View>
  </View>
</View>
`;
